src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -554,17 +554,17 @@ public function nodeName()
     /**
      * Returns the node value of the first node of the list.
      *
-     * @param mixed $default When provided and the current node is empty, this value is returned and no exception is thrown
+     * @param string|null $default When not null: the value to return when the current node is empty
      *
      * @return string The node value
      *
      * @throws \InvalidArgumentException When current node is empty
      */
-    public function text(/* $default = null */)
+    public function text(/* string $default = null */)
     {
         if (!$this->nodes) {
-            if (0 < \func_num_args()) {
-                return func_get_arg(0);
+            if (0 < \func_num_args() && null !== func_get_arg(0)) {
```

Comment:
I am not sure about this change. This will make the ability to pass `null` effectively useless. Is that intended? Should we change the default value to the empty string instead?

Reply:
If we allow `null` as a return type, it means we have to declare the method as returning `?string`. IDE/static analyzers will complain that the null is not checked while there's no need to. But if returning `null` is really normal, we should do it, that's what I'm asking basically.

(We cannot set the default value to the empty string, not sure why you suggest so - it'd mean "don't throw")

Reply:
> (We cannot set the default value to the empty string, not sure why you suggest so - it'd mean "don't throw")

The current behaviour behaviour is to never throw as soon as the user passes a value. Changing the default value wouldn't influence that but only what is returned. If this change is intended as a bugfix for the previous behaviour, then I think it's worth it to make this change.

Reply:
OK, got it. Then here is another consideration: having a method that changes its behavior based on the number of passed arguments is not that a nice API to me.
e.g. `substr($s, 0, null)` doesn't work as expected so you're forced to write an if+else if you want to not specify the length (note that mb_substr() accepts null to mean "up to the end")

I'm fine both ways, but I want us to have this discussion also :)

Reply:
your reasoning makes sense to me